apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ValueConversion.cs

```diff
@@ -180,24 +176,25 @@ public static Value ToValue(object value, SpannerDbType spannerDbType)
                 case TypeCode.Bool:
                     return new Value {BoolValue = Convert.ToBoolean(value)};
                 case TypeCode.String:
+                    if (value is DateTime)
+                    {
+                        // If the value is a DateTime, we always convert using XmlConvert.
+                        // This allows us to convert back to a datetime reliably from the
+                        // resulting string (so roundtrip works properly if the developer uses
+                        // a string as a backing field for a datetime for whatever reason).
+                        return new Value { StringValue = XmlConvert.ToString((DateTime)value, XmlDateTimeSerializationMode.Utc) };
+                    }
+                    return new Value { StringValue = Convert.ToString(value, CultureInfo.InvariantCulture) };
                 case TypeCode.Int64:
-                    return new Value {StringValue = value.ToString()};
+                    return new Value {StringValue = Convert.ToInt64(value).ToString(CultureInfo.InvariantCulture) };
```

Comment:
We don't know what type `value` is here, right? If it's a string, we should be using the invariant culture here, too.
(This is one thing I dislike about the `Convert` methods - they make it unclear whether we're performing string parsing, string formatting, or just a numeric/bitwise conversion such as `long` to `byte`.)